src/App/Serialization/JsonRpcResponseErrorNormalizer.php

```diff
@@ -0,0 +1,151 @@
+<?php
+namespace Yoanm\JsonRpcServer\App\Serialization;
+
+use Yoanm\JsonRpcServer\Domain\Exception\JsonRpcExceptionInterface;
+
+/**
+ * JsonRpcResponseErrorNormalizer prepares response data for the "unexpected" errors occur during request processing.
+ *
+ * It handles "internal server error" appearance in the response.
+ * Instance of this class should be attached to {@see \Yoanm\JsonRpcServer\App\Serialization\JsonRpcResponseNormalizer} only in "debug" mode,
+ * since it will expose vital internal information to the API consumer.
+ *
+ * @see \Yoanm\JsonRpcServer\App\Serialization\JsonRpcResponseNormalizer::normalizeError()
+ */
+class JsonRpcResponseErrorNormalizer
+{
+    /**
+     * @var int maximum count of trace lines to be displayed.
+     */
+    private $maxTraceSize;
+
+    /**
+     * @var bool whether to show trace arguments.
+     */
+    private $showTraceArguments;
+
+    /**
+     * @var bool whether to simplify trace arguments representation.
+     */
+    private $simplifyTraceArguments;
+
+    /**
+     * @param int $maxTraceSize maximum count of trace lines to be displayed.
+     * @param bool $showTraceArguments whether to show trace arguments.
+     * @param bool $simplifyTraceArguments whether to simplify trace arguments representation.
+     */
+    public function __construct(int $maxTraceSize = 10, bool $showTraceArguments = true, bool $simplifyTraceArguments = true)
+    {
+        $this->maxTraceSize = $maxTraceSize;
+        $this->showTraceArguments = $showTraceArguments;
+        $this->simplifyTraceArguments = $simplifyTraceArguments;
+    }
+
+    /**
+     * @param JsonRpcExceptionInterface $error
+     * @return array
+     */
+    public function normalize(JsonRpcExceptionInterface $error) : array
+    {
+        return $this->composeDebugErrorData($error->getPrevious() ?? $error);
```

Comment:
What's the purpose of skipping original error if there is a previous one ?

Reply:
Here we create a debug info about an exception occured.

`JsonRpcExceptionInterface` mainly serves as a wrapper around raw exception. See:
https://github.com/yoanm/php-jsonrpc-server-sdk/blob/v3.2.1/src/App/Creator/ResponseCreator.php#L54-L63

There is no point to display debug for `JsonRpcInternalErrorException` as it will always refer to the same line throwed, has the same class name and so on.

Thus `JsonRpcExceptionInterface` has a little value on its own ragarding debug info composition.
Using previous exception creates much cleaner and useful results.